polyfill/lib/absolute.mjs

```diff
@@ -185,36 +185,39 @@ export class Absolute {
     SetSlot(resultObject, EPOCHNANOSECONDS, { ms: epochMilliseconds, ns: restNanoseconds });
     return this === Absolute ? resultObject : new this(resultObject.getEpochNanoseconds());
   }
-  static fromString(isoString) {
-    isoString = ES.ToString(isoString);
-    const match = STRING.exec(isoString);
-    if (!match) throw new RangeError(`invalid absolute: ${isoString}`);
-    const year = ES.ToInteger(match[1]);
-    const month = ES.ToInteger(match[2]);
-    const day = ES.ToInteger(match[3]);
-    const hour = ES.ToInteger(match[4]);
-    const minute = ES.ToInteger(match[5]);
-    const second = ES.ToInteger(match[6]);
-    const millisecond = ES.ToInteger(match[7]);
-    const microsecond = ES.ToInteger(match[8]);
-    const nanosecond = ES.ToInteger(match[9]);
-    const zone = match[11] || match[10] || 'UTC';
-    const datetime = ES.CastDateTime({
-      year,
-      month,
-      day,
-      hour,
-      minute,
-      second,
-      millisecond,
-      microsecond,
-      nanosecond
-    });
-    const result = datetime.inZone(zone || 'UTC', match[11] ? match[10] : 'earlier');
-    return this === Absolute ? result : new this(result.getEpochNanoseconds());
-  }
-  static from(...args) {
-    const result = ES.CastAbsolute(...args);
+  static from(arg) {
+    let result;
+    if (typeof arg === 'object') {
```

Comment:
we discussed this in IRC; but presumably you don't want `Object('foo')` to fall into this path?

Reply:
@ljharb sorry, I didn't apply those changes yet (hence the draft status of the PR), but I'll go about it as discussed.

Reply:
My (weakly held) intuition is that we shouldn't special-case string wrappers here. If we're doing function overloading distinguishing different kinds of objects, then I'm a bit skeptical of grouping it all into a single `from` method. Overall, I'd suggest just checking of the typeof is "string", if so do the string handling, otherwise Object() it and get the properties.

Reply:
The only precedent I'm aware of for "takes two types of things" involves doing a brand/protocol check (like IsRegExp, or checking if it's an iterator), and then falling back to a naive coercion. In this case, the only object that's branded here is a string primitive/boxed String, so I leaned towards checking for that.

If that seems weird, then having two methods might be better.

Reply:
If we're not throwing for anything with Type() Object/String, I'd be inclined to check Type() == Object and ToString otherwise.

Reply:
I don't think this is analogous to things that take iterators or RegExp-likes as arguments; we're not using an object-oriented protocol. The overloading here is more like some other cases, such as the [TypedArray constructors](https://tc39.es/ecma262/#sec-typedarray-length), which does what @Ms2ger suggested as a check.

Reply:
Sure, but that *only* takes a string; it doesn’t also take a bag of properties like this does.